network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -252,4 +294,18 @@ public String toString() {
         .toString();
     }
   }
+
+  /**
+   * write out the set of registered executors to a file so we can reload them on restart.
+   * You must have a lock on executors when calling this
+   */
+  private void saveRegisteredExecutors() throws IOException {
+    if (db != null) {
+      ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+      ObjectOutputStream out = new ObjectOutputStream(bytes);
+      out.writeObject(executors);
+      out.close();
+      db.put("registeredExecutors".getBytes(Charsets.UTF_8), bytes.toByteArray());
```

Comment:
yeah, I will write one entry per key, sorry I was being a little sloppy to just poc using leveldb.  Marcelo pointed out the issue w/ application removed on the other pr -- I'm porting that change here with a test